### PR TITLE
Testing

### LINKED ISSUE #62: Restoration script fails
The upgrade failed (same as #60 I think), but I can't even restore the pre-upgrade backup. It fails here :
https://github.com/YunoHost-Apps/jellyfin_ynh/blob/63ab03ab79cc088a52b8a36fb4fdf5918bf0e14b/scripts/restore#L94
```

admin@salamandar ~ ??? sudo yunohost backup restore jellyfin-pre-upgrade1                                                                     ? 1 
Info: Preparing archive for restoration...
Info: Restoring jellyfin...
Info: [+...................] > Loading settings...
Info: [#++.................] > Validating restoration parameters...
Info: [###++...............] > Restoring the app main directory...
Info: [#####+..............] > Recreating the dedicated system user...
Info: [######+++++++.......] > Reinstalling dependencies...
Info: '/etc/apt/trusted.gpg.d/jellyfin.asc' wasn't deleted because it doesn't exist.
Info: [#############++.....] > Restoring the systemd configuration...
Error: Could not restore jellyfin
Info: The operation 'Restore 'jellyfin' from a backup archive' could not be completed. Please share the full log of this operation using the command 'yunohost log display 20210312-141854-backup_restore_app-jellyfin --share' to get help
Warning: Here's an extract of the logs before the crash. It might help debugging the error:
Info: DEBUG - + not_mandatory=0
Info: DEBUG - + local archive_path=/home/yunohost.backup/tmp/jellyfin-pre-upgrade1/apps/jellyfin/backup/lib/systemd/system/jellyfin.service
Info: DEBUG - + '[' '!' -d /home/yunohost.backup/tmp/jellyfin-pre-upgrade1/apps/jellyfin/backup/lib/systemd/system/jellyfin.service ']'
Info: DEBUG - + '[' '!' -f /home/yunohost.backup/tmp/jellyfin-pre-upgrade1/apps/jellyfin/backup/lib/systemd/system/jellyfin.service ']'
Info: DEBUG - + '[' '!' -L /home/yunohost.backup/tmp/jellyfin-pre-upgrade1/apps/jellyfin/backup/lib/systemd/system/jellyfin.service ']'
Info: DEBUG - + '[' 0 == 0 ']'
Info: DEBUG - ++ _get_archive_path '"/lib/systemd/system/jellyfin.service"'
Info: DEBUG - ++ python -c '
Info: DEBUG - import sys
Info: DEBUG - import csv
Info: DEBUG - with open(sys.argv[1], '\''r'\'') as backup_file:
Info: DEBUG -     backup_csv = csv.DictReader(backup_file, fieldnames=['\''source'\'', '\''dest'\''])
Info: DEBUG -     for row in backup_csv:
Info: DEBUG -         if row['\''source'\'']==sys.argv[2].strip('\''"'\''):
Info: DEBUG -             print row['\''dest'\'']
Info: DEBUG -             sys.exit(0)
Info: DEBUG -     raise Exception('\''Original path for %s not found'\'' % sys.argv[2])
Info: DEBUG -     ' /home/yunohost.backup/tmp/jellyfin-pre-upgrade1/backup.csv '"/lib/systemd/system/jellyfin.service"'
Info: DEBUG - ++ return 1
Info: DEBUG - + archive_path=/home/yunohost.backup/tmp/jellyfin-pre-upgrade1/
Info: DEBUG - + ynh_exit_properly
Info: [....................] > Loading installation settings...
Info: [+...................] > Stopping and removing the systemd service...
Info: [#+..................] > Removing dependencies...
Info: [##+++++++++++.......] > Removing app main directory...
Info: '/etc/jellyfin' wasn't deleted because it doesn't exist.
Info: [#############+......] > Removing NGINX web server configuration...
Info: [##############++....] > Removing the dedicated system user...
Info: [####################] > Removal of jellyfin completed
```

The expected file is not in `/lib` but in `/usr/lib`.

### LINKED ISSUE #72: Installation issues (10.7.5)
As I said on the other issue, I'm reinstalling Jellyfin from scratch.
And…
```
Info: [##########+.........] > Configuring the settings...
Info: The service jellyfin has correctly executed the action restart.
Warning: sed: can't read /etc/jellyfin/network.xml: No such file or directory
Error: Unable to install jellyfin: An error occurred inside the app installation script
Info: The operation 'Install the 'jellyfin' app' could not be completed. Please share the full log of this operation using the command 'yunohost log share 20210514-075237-app_install-jellyfin' to get help
```

### LINKED ISSUE #74: upnp / dlna
Hi everyone and first a huge thank you for porting Jellyfin under yunohost (special thanks @tituspijean )
I come to you to solve my problem, let me explain.
I am trying to use Jellyfin to play my music on my upnp / dlna Harman Kardon audio amplifier.
First remark I am systematically obliged to retype the command "sudo yunohost firewall allow UDP 1900" each time my server starts up, otherwise jellyfin and my amplifier cannot communicate ..
So after typing this command, my amp sees the Jellyfin server and conversely Jellyfin sees my amp well .. So I can play my music from Jellyfin-web cast on my amp ..
This is where my real problem comes in, at each end of the song I have the beginning of the song which starts over again before finally moving on to the next song, very strange but above all very unpleasant.
 I noticed that the time progression of the songs became Naan: Naan at the end of the song, and I also have some titles which are directly played without a time value which blocks the playback ..
If anyone has an idea to solve this I am interested ..
Hope the solution exists, thanks again ..

### PR DESCRIPTION
**Pull request by tituspijean**
- See #73 
- Adds discovery ports opening
Closes #74  
Closes #72 
Closes #62

### CODE REVIEW

**Comment by tituspijean** [comment]
!testme

File: scripts/install [hunk 1]
```diff
@@ -70,6 +72,28 @@ ynh_script_progression --message="Finding an available port..." --weight=1
 port=$(ynh_find_port --port=8095)
 ynh_app_setting_set --app=$app --key=port --value=$port
 
+if [ $discovery -eq 1 ]; then
+	ynh_script_progression --message="Configuring firewall..." --time --weight=1
+
+	# Open port 1900 for service auto-discovery
+	if [[ $(ynh_port_available --port=1900) -eq 0 ]]; then
```
Comment (kay0u) [inline scripts/install:79]:
```suggestion
	if ynh_port_available --port=1900; then
```

File: scripts/install [hunk 2]
```diff
@@ -70,6 +72,28 @@ ynh_script_progression --message="Finding an available port..." --weight=1
...
+		ynh_exec_warn_less yunohost firewall allow UDP 1900
+		ynh_app_setting_set --app=$app --key=discovery_service --value=1
+	else
+		ynh_print_warn --message="Port 1900 (for service auto-discovery) is not available. Continuing nonetheless."
+		ynh_app_setting_set --app=$app --key=discovery_service --value=0
+	fi
+
+	# Open port 7359 for client auto-discovery
+	if [[ $(ynh_port_available --port=7359) -eq 0 ]]; then
```
Comment (kay0u) [inline scripts/install:88]:
```suggestion
	if ynh_port_available --port=7359; then
```

**Review by kay0u — APPROVED** [review]
Simpler this way

**Comment by tituspijean** [comment]
!testme

**Review by kay0u — COMMENTED** [review]
!

**Comment by tituspijean** [comment]
!testme

**Comment by tituspijean** [comment]
!testme

File: scripts/upgrade [hunk 3]
```diff
@@ -80,6 +87,34 @@ ynh_abort_if_errors
...
+
+        # Open port 7359 for client auto-discovery
+        if ynh_port_available --port=$discovery_client_port; then
+                discovery_client=1
+                ynh_exec_warn_less yunohost firewall allow UDP $discovery_client_port
+        else
+                discovery_client=0
+                ynh_print_warn --message="Port $discovery_client_port (for client auto-discovery) is not available. Continuing nonetheless."
+        fi
+        ynh_app_setting_set --app=$app --key=discovery_service --value=$discovery_client
```
Comment (alexAubin) [inline scripts/upgrade:115]:
```suggestion
        ynh_app_setting_set --app=$app --key=discovery_client --value=$discovery_client
```

File: scripts/install [hunk 4]
```diff
@@ -70,6 +72,30 @@ ynh_script_progression --message="Finding an available port..." --weight=1
...
+
+	# Open port 7359 for client auto-discovery
+	if ynh_port_available --port=$discovery_client_port; then
+		discovery_client=1
+		ynh_exec_warn_less yunohost firewall allow UDP $discovery_client_port
+	else
+		discovery_client=0
+		ynh_print_warn --message="Port $discovery_client_port (for client auto-discovery) is not available. Continuing nonetheless."
+	fi
+        ynh_app_setting_set --app=$app --key=discovery_service --value=$discovery_client
```
Comment (alexAubin) [inline scripts/install:96]:
```suggestion
        ynh_app_setting_set --app=$app --key=discovery_client --value=$discovery_client
```

File: scripts/install [hunk 5]
```diff
@@ -167,7 +200,17 @@ ynh_use_logrotate
 #=================================================
 ynh_script_progression --message="Integrating service in YunoHost..." --weight=1
 
-yunohost service add $app --description="Jellyfin media center" --log_type="systemd"
+# Build the ports list
+needed_ports=()
+(( $discovery_server == 1 )) && needed_ports+=( "$discovery_server_port" )
```
Comment (alexAubin) [inline scripts/install:205]:
```suggestion
(( $discovery_service == 1 )) && needed_ports+=( "$discovery_service_port" )
```

File: scripts/remove [hunk 6]
```diff
@@ -17,6 +17,8 @@ ynh_script_progression --message="Loading installation settings..." --weight=1
 app=$YNH_APP_INSTANCE_NAME
 
 domain=$(ynh_app_setting_get --app=$app --key=domain)
+discovery_client=$(ynh_app_setting_get --app=$app --key=discovery_client)
+discovery_server=$(ynh_app_setting_get --app=$app --key=discovery_server)
```
Comment (alexAubin) [inline scripts/remove:21]:
```suggestion
discovery_service=$(ynh_app_setting_get --app=$app --key=discovery_service)
```

File: scripts/remove [hunk 7]
```diff
@@ -68,6 +70,22 @@ ynh_script_progression --message="Removing logrotate configuration..." --weight=
 # Remove the app-specific logrotate config
 ynh_remove_logrotate
 
+#=================================================
+# CLOSE A PORT
+#=================================================
+
+if [[ $discovery_server -eq 1 ]] && yunohost firewall list | grep -q "\- 1900$"
```
Comment (alexAubin) [inline scripts/remove:77]:
```suggestion
if [[ $discovery_service -eq 1 ]] && yunohost firewall list | grep -q "\- 1900$"
```

File: scripts/restore [hunk 8]
```diff
@@ -51,6 +52,34 @@ fi
...
+
+        # Open port 7359 for client auto-discovery
+        if ynh_port_available --port=$discovery_client_port; then
+                discovery_client=1
+                ynh_exec_warn_less yunohost firewall allow UDP $discovery_client_port
+        else
+                discovery_client=0
+                ynh_print_warn --message="Port $discovery_client_port (for client auto-discovery) is not available. Continuing nonetheless."
+        fi
+        ynh_app_setting_set --app=$app --key=discovery_service --value=$discovery_client
```
Comment (alexAubin) [inline scripts/restore:80]:
```suggestion
        ynh_app_setting_set --app=$app --key=discovery_client --value=$discovery_client
```

File: scripts/restore [hunk 9]
```diff
@@ -102,12 +125,40 @@ ynh_exec_warn_less apt-get -f install $tempdir/jellyfin-ffmpeg.deb -y -o Dpkg::O
...
+
 #=================================================
 # INTEGRATE SERVICE IN YUNOHOST
 #=================================================
 ynh_script_progression --message="Integrating service in YunoHost..." --weight=1
 
-yunohost service add $app --description="Jellyfin media center" --log_type="systemd"
+# Build the ports list
+needed_ports=()
+(( $discovery_server == 1 )) && needed_ports+=( "$discovery_server_port" )
```
Comment (alexAubin) [inline scripts/restore:144]:
```suggestion
(( $discovery_service == 1 )) && needed_ports+=( "$discovery_service_port" )
```

File: scripts/upgrade [hunk 10]
```diff
@@ -185,7 +231,17 @@ ynh_use_logrotate --non-append
 #=================================================
 ynh_script_progression --message="Integrating service in YunoHost..." --weight=1
 
-yunohost service add $app --description="Jellyfin media center" --log_type="systemd"
+# Build the ports list
+needed_ports=()
+(( $discovery_server == 1 )) && needed_ports+=( "$discovery_server_port" )
```
Comment (alexAubin) [inline scripts/upgrade:236]:
```suggestion
(( $discovery_service == 1 )) && needed_ports+=( "$discovery_service_port" )
```

**Review by alexAubin — COMMENTED** [review]
https://www.youtube.com/watch?v=A2VpR8HahKc&list=PLSdoVPM5WnndSQEXRz704yQkKwx76GvPV&index=1

File: scripts/upgrade (hunk 3 above)
Comment (tituspijean) [reply scripts/upgrade:115]: Damn I am dense. Do not code while tired, children.

File: scripts/install (hunk 5 above)
Comment (tituspijean) [reply scripts/install:205]: aaaaaand I was like "noice, `server` and `client` have the same length".

**Comment by tituspijean** [comment]
!testme

**Comment by tituspijean** [comment]
[Test Badge](https://ci.pijean.ovh/ci/job/2)

**Comment by tituspijean** [comment]
[Test Badge](https://ci.pijean.ovh/ci/job/13)

**Comment by tituspijean** [comment]
[Test Badge](https://ci.pijean.ovh/ci/job/14)

**Pull request merged**